connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -298,20 +329,21 @@ protected Map<String, ByteBuffer> performTaskAssignment(String leaderId, long ma
         Map<String, Collection<ConnectorTaskId>> incrementalTaskAssignments =
                 diff(taskAssignments, currentTaskAssignments);
 
+        previousAssignment = computePreviousAssignment(toRevoke, connectorAssignments, taskAssignments, lostAssignments);
+        previousGenerationId = currentGenerationId;
+        previousMembers = memberAssignments.keySet();
+
         log.debug("Incremental connector assignments: {}", incrementalConnectorAssignments);
         log.debug("Incremental task assignments: {}", incrementalTaskAssignments);
 
-        coordinator.leaderState(new LeaderState(memberConfigs, connectorAssignments, taskAssignments));
```

Comment:
I'm not 100% confident that the `connectorAssignments` == `clusterAssignment.allAssignedConnectors()`, and `taskAssignments` == `clusterAssignment.allAssignedTasks()`. Could you help me explain and confirm it?

Reply:
Sure!

Working backwards from the existing code base at this line, we see the call to `coordinator::LeaderState` with the arguments `connectorAssignments` and `taskAssignments`: https://github.com/apache/kafka/blob/2b64f1a57160884754f0a3652c0bd34de21e9ad1/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java#L304

These two arguments are defined earlier on in `performTaskAssignment`: https://github.com/apache/kafka/blob/2b64f1a57160884754f0a3652c0bd34de21e9ad1/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java#L251-L256


In this PR's branch, we see that the `ClusterAssignment` returned from `performTaskAssignment` uses the same `connectorAssignments` and `taskAssignments` as its `allConnectorAssignments` and `allTaskAssignments` fields: https://github.com/C0urante/kafka/blob/b45f73f572ef65b2dbd7c2b73a53a4fb1927f6a4/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java#L344-L345

And these two arguments are defined in the same way and the same place in `performTaskAssignment`: https://github.com/C0urante/kafka/blob/b45f73f572ef65b2dbd7c2b73a53a4fb1927f6a4/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java#L282-L287


Does that help?